### DISCUSSION TITLE
Does Spacedrive will depends on Libraries?

### DISCUSSION THREAD

**Discussion by albertkinng** [spacedriveapp/spacedrive#908] [General]
I have been utilizing [Eagle](https://eagle.cool) for several years now, and it has become an essential tool for me. The only aspect of Eagle that I dislike, and have consequently adapted to, is the file handling using libraries. My approach to using Eagle is simple: I create a folder for each client, and within that folder, I have subfolders for each of their [projects.](https://share.cleanshot.com/2HSCrdNl) These project folders contain all the necessary assets, including fonts, logos, images, notes, color palettes, templates, Adobe files, HTML files, PNGs, PDFs, etc. This system allows me to access everything I need for a project without concern. For instance, if a client requests an image change for their website, I can easily navigate to their folder and double-click on the document file to begin editing.

However, the issue with libraries is that Eagle automatically renames everything under a new hierarchy, making it difficult to access legacy documents that request missing files. I often struggle to locate these files because I do not know the random numeric name that Eagle uses to store them. This situation can turn into a nightmare when dealing with old or legacy documents that need to be opened frequently.

I am curious to know if Spacedrive uses the same file management style as Eagle.

**Comment by jamiepine** [comment]
Thanks for sharing, Spacedrive does not alter the original files in any way, the name stays in tact. 

Spacedrive considers a "Library" to be the database for which all metadata and file system structures are saved. We use "Locations" as places to look for files and continuously watch for changes. It is possible to modify Locations from Spacedrive, or the underlying OS, it will always stay in sync.

Our search spans across your entire Library, including offline Locations. So if a file you needed for a client happened to be on an unplugged hard drive, or a disconnected cloud; we can prompt you to connect that storage Location. Metadata and thumbnails are part of the library so you'll always know what a file is, even if it's offline.

It looks like most of the functionality in Eagle will be matched by Spacedrive. Any features you might need that we've missed early on, we can look into adding so you could make the switch.

**Reply by albertkinng** [reply]
This sounds great! If Spacedrive works as you've described, I'm excited to become a user as soon as it becomes available. Thank you for taking the time to answer my questions. I do have a request though - please consider creating a one-click solution for importing Eagle libraries.
